File: src/codes/Users_Manual/emulator/emulator_voltage.py
from pyautd3 import (
    AUTD3,
    Controller,
    Duration,
    EmitIntensity,
    Phase,
    Silencer,
    Static,
    Uniform,
)
from pyautd3.emulator import Recorder

with Controller.builder([AUTD3([0.0, 0.0, 0.0])]).into_emulator() as emulator:

    def f(autd: Controller[Recorder]) -> Controller[Recorder]:
        autd.send(Silencer.disable())
        autd.send(
            (
                Static.with_intensity(0xFF),
                Uniform((Phase(0x40), EmitIntensity(0xFF))),
            ),
        )
        autd.tick(Duration.from_millis(1))
        return autd

    record = emulator.record(f)

    df = record.output_voltage()
    print(df)

File: src/codes/Users_Manual/emulator/emulator_voltage.rs
# use autd3::prelude::*;
use autd3_emulator::*;
# use std::time::Duration;

# fn main() -> Result<(), Box<dyn std::error::Error>> {
let emulator = Emulator::new([AUTD3::default()]);

let record = emulator.record(|autd| {
    autd.send(Silencer::disable())?;
    autd.send((
        Static { intensity: 0xFF },
        Uniform {
            intensity: Intensity::MAX,
            phase: Phase(0x40),
        },
    ))?;
    autd.tick(Duration::from_millis(1))?;
    Ok(())
})?;

let df = record.output_voltage();
dbg!(df);
#     Ok(())
# }